### PR TITLE
WPCOM MU: Hide the Add-Ons menu on atomic sites

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/dotcom-forge/issues/5839

## Proposed changes:

* Hide the Add-Ons menu on atomic sites

| Before | After |
| - | - |
| image | image |

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Check out PR on a WoA site
* On that WoA site, ensure that your are proxied, go to $site.com/wp-admin
* Verify hosting menu item shows
* Ensure that the Add-Ons menu is gone

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php [hunk 1]
```diff
@@ -87,14 +87,16 @@ function wpcom_add_wpcom_menu_item() {
...
 
-	add_submenu_page(
-		$parent_slug,
-		esc_attr__( 'Add-ons', 'jetpack-mu-wpcom' ),
-		esc_attr__( 'Add-ons', 'jetpack-mu-wpcom' ),
-		'manage_options',
-		esc_url( "https://wordpress.com/add-ons/$domain" ),
-		null
-	);
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
```
Comment (miksansegundo) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php:90]: I see, this is the checking that we can do in https://github.com/Automattic/jetpack/pull/36060

**Review by miksansegundo — DISMISSED** [review]
LGTM 👍

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php (hunk 1 above)
Comment (fushar) [reply projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php:90]: TIL

**Review by fushar — DISMISSED** [review]
Nice job!

**Comment by arthur791004** [comment]
@miksansegundo @fushar I pushed a change to resolve the checks, so I need another approval 😉

**Review by fushar — APPROVED** [review]

**Pull request merged**